dsr/python/ti_files/CatalogFile.py

```diff
@@ -0,0 +1,108 @@
+import os
+import io
+import sys
+import traceback
+import math
+import logging
+from ti_files import ti_files
+from tinames import tinames
+from tifloat import tifloat
+from Pab import *
+
+logger = logging.getLogger(__name__)
+
+class CatalogFile(object):
+
+    def __init__(self, localpath):
+        self.recNum = 0
+        self.localpath = localpath
+
+    @staticmethod
+    def load(path, pab):
+        if mode(pab) == INPUT and dataType(pab) == INTERNAL and recordType(pab) == FIXED:
+            # since it is a directory the recordlength is 38, often it is opened with no value.
+            # TODO: if they specify the longer filename record length, and recordType, then this will be different
+            #       for implementation of long file name handling
+            if recordLength(pab) == 0 or recordLength(pab) == 38:
+                # TODO: load all the directory records upfront
+                return CatalogFile(path)
+
+    def getRecordLength(self):
+        return 38
+
+    def isLegal(self, pab):
+        return mode(pab) == INPUT and dataType(pab) == INTERNAL and recordType(pab) == FIXED
+
+    def readRecord(self, pabRecNum):
+        if pabRecNum != 0:
+            self.recNum = pabRecNum
+        record = self.getRecord(self.recNum)
+        self.recNum += 1
+        return record
+
+    def getRecord(self, idx):
+        if idx == 0:
+            return self.createVolumeData()
+        else:
+            return self.createFileCatRecord()
+
+    def createVolumeData(self):
+        return self.encodeDirRecord("TIPI", 0, 1440, 1438)
+
+    def createFileCatRecord(self):
+        files = sorted(list(filter(lambda x: 

```

Comment:
wastefully listing the entire fileset for every file record... see previous comment.